pkg/cloud/azure/actuators/machine/reconciler.go

```diff
@@ -607,6 +608,11 @@ func (s *Reconciler) createVirtualMachine(ctx context.Context, nicName string) e
 				Namespace: s.scope.Machine.Namespace,
 				Reason:    err.Error(),
 			})
+
+			var detailedError autorest.DetailedError
+			if errors.As(err, &detailedError) && detailedError.Message == "Failure sending request" {
+				return machinecontroller.InvalidMachineConfiguration("failure sending request for machine %s", s.scope.Machine.Name)
+			}
```

Comment:
Are we going to see the actual error logged anywhere when this happens?

Reply:
This error will be returned by reconciler's `Create()`, I'd expect the machine controller to log it somewhere.

Reply:
Ok, wonder if there's any more information than `Failure sending request` that we could extract to improve the UX?